packages/backend/test/mock.ts

```diff
@@ -1,18 +1,31 @@
-export function mock<T>(overrides: Partial<T> = {}): T {
-  const clone = { ...overrides }
+import { Mock as MockFunction, mockFn } from 'earljs'
+
+export type MockedObject<T> = {
+  [P in keyof T]: T[P] extends (...args: any[]) => any
+    ? MockFunction.Of<T[P]>
+    : T[P]
+} & T
+
+export function mock<T>(overrides: Partial<T> = {}): MockedObject<T> {
```

Comment:
Let me know what you think about this.

This function will now turn `{}` overrides into a mock object where every method is a mock function made with Earl's `mockFn`.